### PR TITLE
feat: Adds bypassChecksum to Token constructor to stop checksum validation by default

### PR DESCRIPTION
**Pull request by dalmendray**
Adds the option to not checksum addresses in the Token constructor by default. Alternatively we perform a soft check if the bypassChecksum flag is on.

Soft check: length === 42 && startsWith('0x') && isHex(address)

### CODE REVIEW

File: src/utils/validateAndParseAddress.test.ts [hunk 1]
```diff
@@ -19,3 +19,35 @@ describe('#validateAndParseAddress', () => {
...
+    )
+  })
+
+  it('throws if length > 42', () => {
+    expect(() => checkValidAddress('0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6fA')).toThrow(
+      '0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6fA is not a valid address.'
+    )
+  })
+
+  it('throws if it does not startis with 0x', () => {
```
Comment (JFrankfurt) [inline src/utils/validateAndParseAddress.test.ts:42]: `startis` typo

File: src/utils/validateAndParseAddress.ts [hunk 2]
```diff
@@ -11,3 +11,18 @@ export function validateAndParseAddress(address: string): string {
     throw new Error(`${address} is not a valid address.`)
   }
 }
+
+// Checks a string starts with 0x, is 42 characters long and contains only hex characters after 0x
+const startsWith0xLen42HexRegex = /^(0x)?[0-9a-fA-F]{40}$/
```
Comment (NoahZinsmeister) [inline src/utils/validateAndParseAddress.ts:16]: probably should remove the `?` and group `()`  to enforce the `0x` prefix?

**Review by NoahZinsmeister — COMMENTED** [review]
lgtm pending comments!

File: src/utils/validateAndParseAddress.ts (hunk 2 above)
Comment (dalmendray) [reply src/utils/validateAndParseAddress.ts]: Oops, great catch!

**Review by NoahZinsmeister — APPROVED** [review]

**Pull request merged**